src/ui/api/member.go

```diff
@@ -41,47 +39,82 @@ type memberReq struct {
 
 // Prepare validates the URL and parms
 func (pma *ProjectMemberAPI) Prepare() {
-	pid, err := strconv.ParseInt(pma.Ctx.Input.Param(":pid"), 10, 64)
+	pma.BaseController.Prepare()
+
+	if !pma.SecurityCtx.IsAuthenticated() {
+		pma.HandleUnauthorized()
+		return
+	}
+	user, err := dao.GetUser(models.User{
+		Username: pma.SecurityCtx.GetUsername(),
+	})
 	if err != nil {
-		log.Errorf("Error parsing project id: %d, error: %v", pid, err)
-		pma.CustomAbort(http.StatusBadRequest, "invalid project Id")
+		pma.HandleInternalServerError(
+			fmt.Sprintf("failed to get user %s: %v",
+				pma.SecurityCtx.GetUsername(), err))
 		return
 	}
-	p, err := dao.GetProjectByID(pid)
+	pma.currentUserID = user.UserID
+
+	pid, err := pma.GetInt64FromPath(":pid")
+	if err != nil || pid <= 0 {
+		text := "invalid project ID: "
+		if err != nil {
+			text += err.Error()
+		} else {
+			text += fmt.Sprintf("%d", pid)
+		}
+	}
+	project, err := pma.ProjectMgr.Get(pid)
 	if err != nil {
-		log.Errorf("Error occurred in GetProjectById, error: %v", err)
-		pma.CustomAbort(http.StatusInternalServerError, "Internal error.")
+		pma.HandleInternalServerError(
+			fmt.Sprintf("failed to get project %d: %v", pid, err))
+		return
+	}
+	if project == nil {
+		pma.HandleNotFound(fmt.Sprintf("project %d not found", pid))
+		return
 	}
+	pma.project = project
 
-	if p == nil {
-		log.Warningf("Project with id: %d does not exist.", pid)
-		pma.CustomAbort(http.StatusNotFound, "Project does not exist")
+	if pma.Ctx.Input.IsGet() && !pma.SecurityCtx.HasReadPerm(pid) ||
+		!pma.SecurityCtx.HasAllPerm(pid) {
+		pma.HandleForbidden(pma.SecurityCtx.GetUsername())
+		return
 	}
-	pma.project = p
-	pma.currentUserID = pma.ValidateUser()
-	mid := pma.Ctx.Input.Param(":mid")
-	if mid == "current" {
-		pma.memberID = pma.currentUserID
-	} else if len(mid) == 0 {
-		pma.memberID = 0
-	} else if len(mid) > 0 {
-		memberID, err := strconv.Atoi(mid)
+
+	if len(pma.GetStringFromPath(":mid")) != 0 {
+		mid, err := pma.GetInt64FromPath(":mid")
+		if err != nil || mid <= 0 {
+			text := "invalid member ID: "
+			if err != nil {
+				text += err.Error()
+			} else {
+				text += fmt.Sprintf("%d", mid)
+			}
+			pma.HandleBadRequest(text)
+			return
+		}
+
+		member, err := dao.GetUser(models.User{
```

Comment:
It should be my fault, the member id seems to map to user id?  This doesn't look quite right, could you open an issue for us to follow up?